src/util.c

```diff
@@ -297,6 +297,19 @@ void do_sleep(unsigned int sec)
 		;
 }
 
+void do_usleep(unsigned int usec)
+{
+	struct timespec deadline;
+	clock_gettime(CLOCK_MONOTONIC, &deadline);
+
+	deadline.tv_nsec += usec * 1000;
+	deadline.tv_sec += deadline.tv_nsec / 1000000000;
+	deadline.tv_nsec = deadline.tv_nsec % 1000000000;
+
+	while (clock_nanosleep(CLOCK_MONOTONIC, TIMER_ABSTIME, &deadline, NULL) < 0 && errno == EINTR)
```

Comment:
The clock_nanosleep() API seems to return positive error codes, never negative ones.  Yes, I was also surprised.

Reply:
Arha, did not notice that, nice catch, will fix it in V2.